File: sdk/python-sdk/verity_sdk/protocols/UpdateConfigs.py
from verity_sdk.protocols.Protocol import Protocol
from verity_sdk.utils import EVERNYM_MSG_QUALIFIER


class UpdateConfigs(Protocol):
    MSG_FAMILY = 'update-configs'
    MSG_FAMILY_VERSION = '0.6'

    # Messages
    UPDATE_CONFIGS = 'update'
    GET_STATUS = 'get-status'

    def __init__(self, name=None, logo_url=None):
        super().__init__(self.MSG_FAMILY,
                         self.MSG_FAMILY_VERSION,
                         msg_qualifier=EVERNYM_MSG_QUALIFIER)
        self.name = name
        self.logo_url = logo_url

    def update_msg(self, context):
        msg = self._get_base_message(self.UPDATE_CONFIGS)
        msg['configs'] = [
            {'name': 'name', 'value': self.name},
            {'name': 'logoUrl', 'value': self.logo_url}
        ]

        return msg

    async def update_msg_packed(self, context):
        msg = self.update_msg(context)
        return await self.get_message_bytes(context, msg)

    async def update(self, context):
        return await self.send_message(context, await self.update_msg_packed(context))

    def status_msg(self, _):
        msg = self._get_base_message(self.GET_STATUS)
        return msg

    async def status_msg_packed(self, context):
        return await self.get_message_bytes(context, self.status_msg(context))

    async def status(self, context):
        await self.send_message(context, await self.status_msg_packed(context))

File: sdk/nodejs-sdk/src/protocols/v0_6/UpdateConfigs.js
'use strict'
const utils = require('../../utils')
const Protocol = require('../Protocol')

/**
 * An interface for controlling a 0.6 UpdateConfigs protocol.
 * @extends Protocol
 */
class UpdateConfigs extends Protocol {
  /**
   * Constructor for the 0.6 UpdateEndpoint object. This constructor creates an object that is ready to update
   * the endpoint.
   *
   * @param name Organization's name that is presented to third-parties
   * @param logoUrl A url to a logo that is presented to third-parties
   * @return 0.6 UpdateConfigs object
   *
   * @property {String} msgFamily - 'update-configs'
   * @property {String} msgFamilyVersion - '0.6'
   * @property {String} msgQualifier - 'Community Qualifier'
   * @property {String} this.msgNames.UPDATE_CONFIGS - 'update'
   * @property {String} this.msgNames.GET_STATUS - 'get-status'
   */
  constructor (name = null, logoUrl = null) {
    const msgFamily = 'update-configs'
    const msgFamilyVersion = '0.6'
    const msgQualifier = utils.constants.EVERNYM_MSG_QUALIFIER
    super(msgFamily, msgFamilyVersion, msgQualifier)

    this.name = name
    this.logoUrl = logoUrl

    this.msgNames.UPDATE_CONFIGS = 'update'
    this.msgNames.GET_STATUS = 'get-status'
  }

  /**
     * Creates the control message without packaging and sending it.
     * @param context an instance of the Context object initialized to a verity-application agent
     * @return the constructed message (JSON object)
     *
     * @see #updateMsg
     */
  async updateMsg () {
    var msg = this._getBaseMessage(this.msgNames.UPDATE_CONFIGS)
    msg.configs = [
      { name: 'name', value: this.name },
      { name: 'logoUrl', value: this.logoUrl }
    ]

    return msg
  }

  /**
     * Creates and packages message without sending it.
     * @param context an instance of the Context object initialized to a verity-application agent
     * @return the byte array ready for transport
     *
     * @see #updateMsg
     */
  async updateMsgPacked (context) {
    return this.getMessageBytes(context, await this.updateMsg())
  }

  /**
     * Directs verity-application to update the configuration register with the verity-application
     *
     * @param context an instance of the Context object initialized to a verity-application agent
     */
  async update (context) {
    await this.sendMessage(context, await this.updateMsgPacked(context))
  }

  /**
     * Creates the control message without packaging and sending it.
     * @param context an instance of the Context object initialized to a verity-application agent
     * @return the constructed message (JSON object)
     *
     * @see #status
     */
  async statusMsg (context) {
    var msg = this._getBaseMessage(this.msgNames.GET_STATUS)
    return msg
  }

  /**
     * Creates and packages message without sending it.
     * @param context an instance of the Context object initialized to a verity-application agent
     * @return the byte array ready for transport
     *
     * @see #status
     */
  async statusMsgPacked (context) {
    return this.getMessageBytes(context, await this.statusMsg(context))
  }

  /**
     * Ask for status from the verity-application agent
     *
     * @param context an instance of the Context object initialized to a verity-application agent
     */
  async status (context) {
    await this.sendMessage(context, await this.statusMsgPacked(context))
  }
}
module.exports = UpdateConfigs
